### PR TITLE
sortedtable refactoring: remove sortabletable, simplify sortedtable

### PR DESCRIPTION
**Pull request by elkmaster**
Originally we had 2 components for table functionality `sortabletable` and `sortedtable`. `sortabletable` contains mostly all code, `sortedtable` is just wrap it and adds couple additional api's. Through admin-ui codebase everywhere used wrapper version of the table. Only in one, seems to be very old, part  of the code is used the table without wrapper.

In order to have single version of it, I moved all code in one place, under `sortedtable` component, as it is used everywhere. Another issue was that all code lived in one file, so tried to split it in smaller components (<Head>, <Row>, <Spinner>) with corresponding css in modular style.

Overall types\props consumed by refactored table not changed, so integrate it in admin-ui should be as easy as just change import statements. 

And also this pr contains latests updates from [CRDB-Design-System_V2](https://www.figma.com/file/B5AtEGdRbW3VxiBdg62TB8Xw/CRDB-Design-System_V2?node-id=0%3A1) (fonts, colors, paddings).

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/admin-ui-components/22)

### CODE REVIEW

File: src/sortedtable/sortedtable.tsx [hunk 1]
```diff
@@ -99,12 +110,108 @@ interface SortedTableState {
...
+ * 'onChangeSortSetting' callback property.
+ */
+
 export class SortedTable<T> extends React.Component<
   SortedTableProps<T>,
   SortedTableState
 > {
   static defaultProps: Partial<SortedTableProps<any>> = {
     rowClass: (_obj: any) => "",
+    count: 0,
```
Comment (koorosh) [inline src/sortedtable/sortedtable.tsx:208]: `count` value is calculated inside of the component (LINE 369). Do we need `count` prop or it can be removed?

File: src/sortedtable/utils.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { createSelector } from "reselect";
```
Comment (koorosh) [inline src/sortedtable/utils.js:1]: Probably it has to be typescript (`utils.ts`) file?

File: src/sortedtable/sortedtable.tsx [hunk 3]
```diff
@@ -99,12 +110,108 @@ interface SortedTableState {
...
+export interface SortSetting {
+  sortKey: any;
+  ascending: boolean;
+  columnTitle?: string;
+}
+
+/**
+ * TableProps are the props that should be passed to SortableTable.
+ */
+interface TableProps {
```
Comment (koorosh) [inline src/sortedtable/sortedtable.tsx:154]: `TableProps` interface is not used anymore.

File: src/sortedtable/sortedtable.tsx [hunk 4]
```diff
@@ -82,6 +92,7 @@ interface SortedTableProps<T> {
   // empty state for table
   empty?: boolean;
```
Comment (koorosh) [inline src/sortedtable/sortedtable.tsx:93]: Do you think we need `empty` prop so far? It looks like, with current changes, it's possible to calculate if there is some data to display or not.

**Review by koorosh — COMMENTED** [review]
In general, looks good! Just several comments to take a look at (mentioned above)

File: .gitignore [hunk 5]
```diff
@@ -1,2 +1,3 @@
 node_modules
 dist
+.DS_Store
```
Comment (koorosh) [inline .gitignore:3]: New line is required at the end of file

File: webpack.config.js [hunk 6]
```diff
@@ -17,9 +17,13 @@ module.exports = {
...
   resolve: {
+    modules: [
+      'node_modules',
+      path.join(__dirname, 'src/fonts'),
+    ],
     extensions: [".ts", ".tsx", ".js", ".jsx", ".less"],
     alias: {
-      src: path.resolve(__dirname, "src")
+      src: path.resolve(__dirname, "src"),
     },
```
Comment (dhartunian) [inline webpack.config.js:27]: Why are these changes necessary?

**Review by dhartunian — COMMENTED** [review]
@elkmaster can you rebase this on master and address the changes so we can get it merged? This is pre-requisite for the Transactions page PR I assume.

File: src/sortedtable/sortedtable.tsx (hunk 4 above)
Comment (elkmaster) [reply src/sortedtable/sortedtable.tsx]: In admin-ui this prop is used heavily, I would leave as is at least for this iteration of refactoring.

**Review by koorosh — COMMENTED** [review]
In general looks great! 
Found one place with broken styles:
- text in rows by default has `mono-space` fonts, even in cells that should be with default font. The last column `Diagnostics` inherits font styles and has mono-space font.

**Comment by elkmaster** [comment]
> * text in rows by default has `mono-space` fonts

thanks, pushed fix

**Review by dhartunian — COMMENTED** [review]
@elkmaster the commit message for this PR needs to have a comprehensive summary of what is being changed. I would like to see a list of what files, types etc. were moved, deleted, renamed. It's not possible to review this PR without understanding your intent.

> Reviewed 5 of 27 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/admin-ui-components/22#-:-MJ-cxRU7BZlDYs6kfi_:brjvqet)* status: 5 of 27 files reviewed, all discussions resolved

**Comment by koorosh** [comment]
@elkmaster  Noticed one visual regression, background color for table headers is white now instead
Screen Shot 2020-11-20 at 1 40 47 PM

**Review by koorosh — CHANGES_REQUESTED** [review]
Looks good! The only thing to correct is background color for headers (my previous comment)

**Comment by elkmaster** [comment]
@koorosh, thanks for the review! You are right but partially, header is wrong but by fonts. It should be semibold and capitalized. Pushed fix for that. Take a look in figma please: https://www.figma.com/file/B5AtEGdRbW3VxiBdg62TB8Xw/CRDB-Design-System?node-id=0%3A1

Screenshot 2020-11-26 at 17 58 31

**Review by koorosh — APPROVED** [review]
Great, you're right! 👍🏽

> Reviewed 13 of 27 files at r1, 11 of 12 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/admin-ui-components/22#-:-MN8rk4PDp9az_tUjKC5:bxwgcv7)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @elkmaster)

**Review by dhartunian — APPROVED** [review]
:lgtm: 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/admin-ui-components/22#-:-MO3LcWP4ED_YMnG3hrM:bg7igkv)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @elkmaster)
